Snowflake.Data/Core/Session/CreateSessionTokens.cs

```diff
@@ -0,0 +1,43 @@
+using System;
+using System.Collections.Generic;
+using Snowflake.Data.Core.Session;
+
+namespace Snowflake.Data.Core.Session
+{
+    internal class CreateSessionTokens: ICreateSessionTokens
+    {
+        internal long _timeout { get; set; }  = Timeout;
+        private const long Timeout = 30000; // 30 seconds as default
+        private readonly object _tokenLock = new object();
+        private readonly List<CreateSessionToken> _tokens = new List<CreateSessionToken>();
+        private int _tokenCount = 0;
```

Comment:
my idea was to make Count() operation very fast because Count() will be executed a lot more often than starting or finishing of session creations because it is used for every GetSession() operation sometimes even a few times. So ideally if it was not guarded by any lock. Concurrent collections would be slower on Count().

Reply:
I removed it and I introduced a read write lock